File: API/Dtos/Basket.cs
namespace API.Dtos
{
    public class Basket
    {
        public string UserId { get; set; } 
        public Guid Id { get; set; }
        //追加
        public List<BasketItem> Items { get; set; }

        //public Basket()
        //{
        //    Items = new List<BasketItem>();
        //}
    }
}

File: client/src/app/models/basket.ts
import { Product } from './product';

// export interface BasketItem {
//   productId: number;
//   basketId: string; //guid
//   quantity: number;
//   product: Product[];
// }

// export interface Basket {
//   userId: string;
//   id: string; //guid
//   items: BasketItem[];
// }

// export interface BasketConfirm {
//   basketId: string;
//   quantity: number;
//   product: Product[];
//   // product: Product;
// }
export interface BasketItem {
  productId: number;
  name: string;
  price: number;
  pictureUrl: string;
  brand: string;
  type: string;
  quantity: number;
}

export interface Basket {
  id: number;
  buyerId: string;
  items: BasketItem[];
}
